### ISSUE TITLE
Semantic snippets: Inline statement snippets are not present if completion session is not triggered explicitly

### ISSUE THREAD

**Issue by DoctorKrolic** [dotnet/roslyn#75072]
**Version Used**:
Latest `main`

**Steps to Reproduce**:
```cs
class C
{
    void M(int intVar)
    {
        intVar.$$
    }
}
```
Start typing `for` at position of `$$`.

**Expected Behavior**:
`for` and `forr` semantic snippets are shown

**Actual Behavior**:
No semantic snippets are shown in completion list

**Demo:**
![snippetBug](https://github.com/user-attachments/assets/ec6a37a7-20da-4185-b668-c2b04936d296)
